File: Module 1-CoreJava/28.7.22/Practice/OverridingEx.java
package polymorphism287;

class Bank
{
	int rate()
	{
		return 0;
	}
}
class Sbi extends Bank
{
	int rate()
	{
		return 7;
	}
}
class Icici extends Bank
{
	int rate()
	{
		return 8;
	}
}
class Bob extends Bank
{
	int rate()
	{
		return 6;	
	}
}
class Hdfc extends Bank
{
	int rate()
	{
		return 5;
	}
}
public class OverridingEx 
{
	public static void main(String[] args) 
	{
		Bank b;
		b=new Sbi();
		System.out.println(b.rate());
		b=new Icici();
		System.out.println(b.rate());
		b=new Hdfc();
		System.out.println(b.rate());
		b=new Bob();
		System.out.println(b.rate());
	}
}

File: Module 2-Core Kotlin/26.8/OverridingEx.kt
open class Bank2
{
    open fun rate():Int
    {
        return 0
    }
}
class Sbi :Bank2()
{
    override fun rate():Int
    {
        return 6
    }
}
class Icici:Bank2()
{
    override fun rate():Int
    {
        return 7
    }
}
class Axis:Bank2()
{
    override fun rate():Int
    {
        return 8
    }
}
fun main()
{
    var b=Bank2()
    b=Sbi()
    println(b.rate())
    b=Icici()
    println(b.rate())
    b=Axis()
    println(b.rate())
}